### ISSUE TITLE
Baseline Update - Preview Changes

### ISSUE THREAD

**Issue by c9mbundy** [umbraco/Umbraco.Cloud.Issues#341]
The portal Baseline Update process is either run or don't run, with no option to preview the changes that will be applied if a baseline update is run for a child site - meaning it's very much a leap of faith that it's the right thing to do.

I often need to make changes to a child before applying the baseline update, to ensure I don't break the site and then have it off-line while I fix it, and so having the ability to preview the changes before they are applied would provide a useful check to help make sure I don't forget.

I suspect that a checkbox or button that runs something along these lines might do the trick.

Kudo:: C:\home\site\repository>git merge --no-commit --no-ff upstream/master
Kudo:: C:\home\site\repository>git diff --name-only
Kudo:: C:\home\site\repository>git reset --hard

Basically, the intention is:  run a merge with --no-commit, to pull in changes from the master as "wip", then get the diff and list the files changed, then discard all the "wip" so no changes are made to the child.

However, I'm definitely no git expert, and so haven't actually taken the plunge with this, to see if it will do the job.

**Comment by c9mbundy** [comment]
This git commands above are not a working solution. I know what I'm after, but not sure how to get there.

I'm guessing that the git merge assumes the branch already exists from the baseline.

I'm guessing I'd need to fetch upstream and checkout master before the merge -  but my lack of git expertise means I'm way out of my depth at this point.

**Comment by sitereactor** [comment]
I think you could fetch from the upstream first and then do a diff between the two branches to see the difference and what would be merged into master.
`git diff master..upstream`

**Comment by c9mbundy** [comment]
Thanks for that pointer - So something like:

```
git fetch upstream
get diff master..upstream
```

Sounds like it wouldn't create uncommitted changes, avoiding the need of the git reset ?

**Comment by sitereactor** [comment]
No, nothing new is committed and nothing is merged. You are pulling down changes from the upstream in it's own branch, which will then allow you to diff (or compare) the two branches BEFORE anything is merged. And the merge would be what the Baseline Update button does (more or less).

**Comment by c9mbundy** [comment]
Excellent - I'll give that a go - many thanks.

**Comment by c9mbundy** [comment]
Dang! - spoke too soon.
I know this is not a git tutorial site, so it's my problem to solve, but fetching the upstream never seems to complete - just waits forever.

**Comment by sitereactor** [comment]
Are you doing this through kudu? Kudu is not really appropriate for this. Some things will work okay from the console, but not everything.
You can the same in a local clone by adding a remote to the Baseline master and call it upstream. It's essentially the same as Forking a repository on GitHub and then updating the forked copy from the upstream (the origini repository).

**Comment by sitereactor** [comment]
Docs: https://docs.github.com/en/github/collaborating-with-issues-and-pull-requests/configuring-a-remote-for-a-fork

And for syncing:
https://docs.github.com/en/github/collaborating-with-issues-and-pull-requests/syncing-a-fork
If you only want to see the diff/compare then don't merge as shown in this doc.

**Comment by c9mbundy** [comment]
Yes, I was using Kudu - I'll play with the local installation instead - again many thanks.

Of course, going back to the start of this thread, I would think it still a nice feature if this could be viewed on the Baseline Updater panel, so you can easily see what you are going to push to a client if you update it from the baseline.

**Comment by sitereactor** [comment]
Yes, agree. I would suggest submitting a feature request for "Previewing Baseline updates prior to updating", so we can capture that.
